delphi/app/routes.py

```diff
@@ -1,6 +1,23 @@
 from app import app
+from flask import request
+import cv2
+import numpy
+
 
 @app.route('/')
 @app.route('/index')
 def index():
-    return 'Hello, World!'
\ No newline at end of file
+    return 'Hello, World!'
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    if request.headers['content-length'] != '0' and request.headers['content-type'] in ['image/jpeg', 'image/png']:
+        buf = numpy.frombuffer(request.data, numpy.uint8)
+        current_image = cv2.imdecode(buf, cv2.IMREAD_COLOR)
```

Comment:
Grabs buffer from the payload and converts it into a numpy array which is then decoded into an image/matrix. Do we plan on sending images with or without color?

Reply:
I have only experimented with colored images. Not sure what tradeoffs we would get without color.